src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -209,10 +209,46 @@ public SNIHandle CreateConnectionHandle(object callbackObject, string fullServer
                 case TdsEnums.TCP:
                     return ConstructTcpHandle(serverNameParts[1], timerExpire, callbackObject, parallel);
 
+                case TdsEnums.NP:
+                    if (parallel)
+                    {
+                        SNICommon.ReportSNIError(SNIProviders.INVALID_PROV, 0, SNICommon.MultiSubnetFailoverWithNonTcpProtocol, string.Empty);
+                        return null;
+                    }
+
+                    // Named Pipe Format: np:\\<host name>\pipe\<pipe name>
+                    const int minPipeLength = 10; // eg: //./pipe/a
```

Comment:
NP: Can the pipe path parsing and server name extraction be extracted to another method?